drivers/firmware/nrf_ironside/CMakeLists.txt

```diff
@@ -2,5 +2,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
+zephyr_library_property(ALLOW_EMPTY TRUE)
```

Comment:
Can't we make the inclusion conditional somehow? Instead of adding an empty library to each build.

Reply:
I think a (hidden) symbol should be introduced

```
config NRF_IRONSIDE
	bool
```

And then change `drivers/CMakeLists.txt` with a conditional include

```diff
diff --git a/drivers/firmware/CMakeLists.txt b/drivers/firmware/CMakeLists.txt
index 167d21addb5..8380b3f08c5 100644
--- a/drivers/firmware/CMakeLists.txt
+++ b/drivers/firmware/CMakeLists.txt
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0

 # zephyr-keep-sorted-start
-add_subdirectory(nrf_ironside)
+add_subdirectory_ifdef(CONFIG_NRF_IRONSIDE nrf_ironside)
 add_subdirectory_ifdef(CONFIG_ARM_SCMI scmi)
 # zephyr-keep-sorted-stop
```